src/thread.cr

```diff
@@ -135,4 +135,42 @@ class Thread
       Fiber.inactive(fiber)
     end
   end
+
+  private def stack_address : Void*
+    address = Pointer(Void).null
+
+    {% if flag?(:darwin) %}
+      address = LibC.pthread_get_stackaddr_np(@th) - pthread_get_stacksize_np(@th)
```

Comment:
In https://github.com/crystal-lang/crystal/pull/6970#issuecomment-431652079 you said that these functions return bogus values. Any new insight regarding this?

Reply:
Urgh, I totally forgot about #6970 :/

To be honest, I don't know. Rust does just that, and doesn't care to check the main thread (when it does for OpenBSD). BDWGC doesn't need `pthread_get_stacksize_np`. I'm not sure what should be done, here.

Reply:
It seems it was fixed in macOS 10.10

Reply:
Here is how openjdk does it:
http://hg.openjdk.java.net/jdk8/jdk8/hotspot/file/87ee5ee27509/src/os_cpu/bsd_x86/vm/os_bsd_x86.cpp#l836

Reply:
Ok, from https://github.com/rust-lang/rust/issues/43347#issuecomment-316783599 and https://github.com/dmlloyd/openjdk/blob/2f680fec58a8ffb818a48f542e493306d1113c69/src/hotspot/os_cpu/bsd_x86/os_bsd_x86.cpp#L902-L961 lets move forward.

Also, the osx binaries are targeted for `>= 10.11`.

Reply:
Sadly I can't just take their solution, it's licensed under the GPLv2.